### PR TITLE
Add consensus tests

### PR DESCRIPTION
**Pull request by MakisChristou**
This PR closes [this issue](https://github.com/vechainfoundation/protocol-board-repo/issues/27) by increasing the coverage of all files above 80%

### CODE REVIEW

File: block/block.go [hunk 1]
```diff
@@ -33,6 +36,42 @@ type Body struct {
 	Txs tx.Transactions
 }
 
+// GetMockBlock creates and returns a mock block for testing purposes.
```
Comment (libotony) [inline block/block.go:39]: Please avoid adding test codes to the package source.

File: block/block.go [hunk 2]
```diff
@@ -33,6 +36,42 @@ type Body struct {
...
+func GetMockBlock(parentId thor.Bytes32) *Block {
+	tx1 := new(tx.Builder).Clause(tx.NewClause(&thor.Address{})).Clause(tx.NewClause(&thor.Address{})).Build()
+	tx2 := new(tx.Builder).Clause(tx.NewClause(nil)).Build()
+
+	privKey := string("dce1443bd2ef0c2631adc1c67e5c93f13dc23a41c18b536effbbdcbcdb96fb65")
```
Comment (libotony) [inline block/block.go:44]: Use `genesis.DevAccounts()`

File: block/block.go (hunk 1 above)
Comment (libotony) [inline block/block.go:39]: Take a look at https://github.com/vechain/thor/blob/master/consensus/consensus_test.go#L196

**Review by libotony — CHANGES_REQUESTED** [review]

File: block/block.go (hunk 1 above)
Comment (MakisChristou) [reply block/block.go:39]: I am not sure I understand the suggestion. Should I remove `GetMockBlock` entirely or just move it under tests?

File: block/block.go (hunk 2 above)
Comment (MakisChristou) [reply block/block.go:44]: Also here afaik `genesis.DevAccounts()` just returns a list of addresses but we need a mock block instead. What am I missing here?

Comment (libotony) [reply block/block.go:44]: Yes, it's a list, you can pick any of them

**Comment by MakisChristou** [comment]
I have made some changes can you take a look @libotony ?

File: consensus/consensus_test.go [hunk 3]
```diff
@@ -215,6 +215,52 @@ func (tc *testConsensus) consent(blk *block.Block) error {
...
+	mockStater := &state.Stater{}
+	mockForkConfig := thor.ForkConfig{}
+
+	// Create a new consensus instance
+	consensus := New(mockRepo, mockStater, mockForkConfig)
+
+	// Assert that the consensus instance is not nil
+	if consensus == nil {
+		t.Errorf("Failed to create new consensus instance")
+	}
```
Comment (libotony) [inline consensus/consensus_test.go:230]:
```suggestion
	// Assert that the consensus instance is not nil
	assert.NotNil(t, consensus, "Failed to create new consensus instance")
```

File: consensus/consensus_test.go [hunk 4]
```diff
@@ -215,6 +215,52 @@ func (tc *testConsensus) consent(blk *block.Block) error {
...
+}
+
+func TestNewRuntimeForReplay(t *testing.T) {
+
```
Comment (libotony) [inline consensus/consensus_test.go:234]:
```suggestion
```

File: consensus/consensus_test.go [hunk 5]
```diff
@@ -215,6 +215,52 @@ func (tc *testConsensus) consent(blk *block.Block) error {
...
+	consensus, err := newTestConsensus()
+	b1 := consensus.parent
+
+	// Test for success scenario
+	runtime, err := consensus.con.NewRuntimeForReplay(b1.Header(), false)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, runtime)
+
```
Comment (libotony) [inline consensus/consensus_test.go:243]:
```suggestion
```

File: consensus/consensus_test.go [hunk 6]
```diff
@@ -288,6 +334,8 @@ func TestValidateBlockHeader(t *testing.T) {
 				err = tc.consent(blk)
 				expected := errFutureBlock
 				assert.Equal(t, expected, err)
+				IsFutureBlockError := IsFutureBlock(expected)
+				assert.True(t, IsFutureBlockError)
```
Comment (libotony) [inline consensus/consensus_test.go:338]:
```suggestion
				assert.True(t, IsFutureBlock(expected))
```

File: consensus/consensus_test.go [hunk 7]
```diff
@@ -306,6 +354,10 @@ func TestValidateBlockHeader(t *testing.T) {
 					),
 				)
 				assert.Equal(t, expected, err)
+				isCriticalError := IsCritical(err)
+				assert.True(t, isCriticalError)
+				stringError := err.Error()
+				assert.Equal(t, stringError, "block gas limit invalid: parent 10000000, current 20000000")
```
Comment (libotony) [inline consensus/consensus_test.go:360]:
```suggestion
				assert.True(t, IsCritical(err))
				assert.Equal(t, err.Error(), "block gas limit invalid: parent 10000000, current 20000000")
```

**Review by libotony — COMMENTED** [review]
A few improvement advice.

**Review by libotony — APPROVED** [review]

**Pull request merged**